cmake/compiler.cmake

```diff
@@ -1,37 +1,53 @@
+option( ENABLE_SANITIZERS "Enable sanitizers" OFF )
+if( ENABLE_SANITIZERS )
+  if( CMAKE_CXX_COMPILER_ID MATCHES "Clang" )
+    message( STATUS "Enabling Clang sanitizers..." )
+    add_compile_options( -fsanitize=address,undefined -fno-omit-frame-pointer )
+    add_link_options( -fsanitize=address,undefined )
+  elseif( MSVC )
+    # Reference: https://learn.microsoft.com/en-us/cpp/build/reference/fsanitize
+    message( WARNING "Enabling MSVC sanitizers (-T ClangCL has better support)..." )
+    add_compile_options( /fsanitize=address )
+    add_link_options( /INCREMENTAL:NO )
+  else()
+    message ( FATAL_ERROR "Unsupported compiler for sanitizers: ${CMAKE_CXX_COMPILER_ID}" )
+  endif()
+endif()
+
+# C++ standard
+set( CMAKE_CXX_STANDARD 17 )
+set( CMAKE_CXX_STANDARD_REQUIRED ON )
+set( CMAKE_CXX_EXTENSIONS ON )
 
 if( UNIX )
-  set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -fPIC" )
-  set( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} -O3")
-else()
-  add_definitions( -D_WIN32_WINNT=0x601 )
-  add_definitions( -D_CRT_SECURE_NO_WARNINGS )
-  add_definitions( -DNOMINMAX )
-      
-  set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHc" )
```

Comment:
This flag is generally weird, the default should be used unless there is a compelling reason.